### PR TITLE
rfc: coercion-agent

### PR DESCRIPTION
**Pull request by SchrodingerZhu**

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
this does not solve the deep coercion yet.

File: base/src/main/java/org/aya/tyck/tycker/CoercionAgent.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    new Coercer.PathToPi()
+  );
+
+  static public class CoercionFailure extends Throwable {
+    static CoercionFailure INSTANCE = new CoercionFailure();
+
+    @Override
+    public Throwable fillInStackTrace() {
+      return this;
+    }
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/tycker/CoercionAgent.java:33]: Why?

**Review by ice1000 — COMMENTED** [review]
Is the throwable object intentionally written like that?

**Review by ice1000 — COMMENTED** [review]
I also think you don't have to comment out the code. You can just delete it.

**Comment by SchrodingerZhu** [comment]
Close for further development.

**Pull request closed without merging**